delta-api/src/main/java/io/cdap/delta/api/assessment/TableAssessorSupplier.java

```diff
@@ -33,4 +33,17 @@
    * @throws Exception if the table assessor could not be created
    */
   TableAssessor<T> createTableAssessor(Configurer configurer) throws Exception;
+
+  /**
+   * Create a table assessor that will check if there will be potential problems replicating a table.
+   *
+   * @param configurer configurer used to instantiate plugins
+   * @param tempResourceIdPrefix the prefix of the id for some temporary resources created by Assessor. With the
+   *                             prefix, it will be easy to detect/cleanup dangling resources
+   * @return a table assessor
+   * @throws Exception if the table assessor could not be created
+   */
+  default TableAssessor<T> createTableAssessor(Configurer configurer, String tempResourceIdPrefix) throws Exception {
```

Comment:
instead of adding a new method, it would be better to expose methods like getNamespace(), getName() in the Configurer interface

Reply:
thanks for the suggestion :) done :)